crates/libcontainer/src/rootfs/utils.rs

```diff
@@ -131,51 +131,53 @@ pub fn parse_mount(m: &Mount) -> std::result::Result<MountOptionConfig, MountErr
+                    MountOption::Dev(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Nodev(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Exec(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Noexec(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Sync(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Async(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Dirsync(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Remount(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Mand(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Nomand(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Atime(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Noatime(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Diratime(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Nodiratime(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Bind(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Rbind(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Unbindable(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Runbindable(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Private(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Rprivate(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Shared(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Rshared(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Slave(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Rslave(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Relatime(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Norelatime(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Strictatime(is_clear, flag) => Some((is_clear, flag)),
+                    MountOption::Nostrictatime(is_clear, flag) => Some((is_clear, flag)),
+                },
+                Err(unknown) => {
                     if unknown == "idmap" || unknown == "ridmap" {
-                        return Err(MountError::UnsupportedMountOption(unknown.to_string()));
+                        return Err(MountError::UnsupportedMountOption(unknown));
                     }
                     None
                 }
             } {
                 if is_clear {
-                    flags &= !flag;
+                    flags.remove(flag);
                 } else {
-                    flags |= flag;
+                    flags.insert(flag);
```

Comment:
It makes it easier for code readers to understand what happens! Thanks.

Reply:
Let me revert this

Reply:
reverted in 33608e3